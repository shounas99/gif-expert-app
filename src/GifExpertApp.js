import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ({defaultCategories = []}) => {

    // const categories = ['Jonas Brothers', 'Joe Jonas', 'Kevin Jonas', 'Nick Jonas', '5SOS','Shawn Mendes'];

    // const [categories, setCategories] = useState(['Jonas Brothers']);
    const [categories, setCategories] = useState([defaultCategories]);
    
    // const handleAdd = () =>{
    //     setCategories([...categories, 'The Little Prince']);
    // }
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr /> 
            <ol>
                { 
                    categories.map(category => (
                        <GifGrid 
                        key = {category}
                        category ={category} />
                    ))
                }
            </ol>
        </>
    )
}

export default GifExpertApp;
