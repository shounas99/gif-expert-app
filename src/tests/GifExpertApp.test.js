import React from 'react';
import GifExpertApp from "../GifExpertApp";
import { shallow } from "enzyme"

describe('Pruebas en <GifExpertApp />', () => {

    test('Debe mostrarse corretamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
        
    })

    test('Mostar lista de categorias', () => {
        const categories = ['Jonas Brothers', 'Joe Jonas']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    })
    
    
})
