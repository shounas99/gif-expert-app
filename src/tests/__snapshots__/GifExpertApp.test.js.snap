// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrarse corretamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={Array []}
      key=""
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Mostar lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={
        Array [
          "Jonas Brothers",
          "Joe Jonas",
        ]
      }
      key="Jonas Brothers,Joe Jonas"
    />
  </ol>
</Fragment>
`;
